File: img/FalaMuriloAndroid/app/src/main/java/com/tairone/falamurilo/models/AppLanguage.kt
package com.tairone.falamurilo.models

import java.util.Locale

enum class AppLanguage(val code: String, val label: String, val flag: String) {
    ENGLISH("en-US",    "English",   "🇺🇸"),
    SPANISH("es-ES",    "Español",   "🇪🇸"),
    PORTUGUESE("pt-BR", "Português", "🇧🇷");

    companion object {
        fun detected(): AppLanguage {
            val lang = Locale.getDefault().language
            return when {
                lang.startsWith("pt") -> PORTUGUESE
                lang.startsWith("es") -> SPANISH
                else -> ENGLISH
            }
        }

        fun fromCode(code: String): AppLanguage =
            entries.firstOrNull { it.code == code } ?: ENGLISH
    }
}

File: FalaMurilo/FalaMuriloIOS/Models/AppLanguage.swift
import Foundation

enum AppLanguage: String, CaseIterable, Identifiable {
    case english = "en-US"
    case spanish = "es-ES"
    case portuguese = "pt-BR"

    var id: String { rawValue }

    var label: String {
        switch self {
        case .english:    return "English"
        case .spanish:    return "Español"
        case .portuguese: return "Português"
        }
    }

    var flag: String {
        switch self {
        case .english:    return "🇺🇸"
        case .spanish:    return "🇪🇸"
        case .portuguese: return "🇧🇷"
        }
    }

    static func detected() -> AppLanguage {
        let code = Locale.current.language.languageCode?.identifier ?? "en"
        if code.hasPrefix("pt") { return .portuguese }
        if code.hasPrefix("es") { return .spanish }
        return .english
    }
}
